app/models/kuroko2/job_instance.rb

```diff
@@ -43,10 +46,14 @@ def cancelable?
     tokens.first.try(:cancelable?)
   end
 
-  def cancel
+  def cancel(by:)
     self.tokens.destroy(*self.tokens)
     self.executions.destroy(*self.executions)
     self.touch(:canceled_at)
+
+    message = "This job was canceled by #{by}."
+    self.logs.warn(message)
+    Kuroko2.logger.warn(message)
   end
```

Comment:
>  confusing behavior for users

agree. And reconsider my opinion, So it changes to enable manual cancellation notifications https://github.com/cookpad/kuroko2/pull/8/commits/01cf47f16a027a9e16fbb07ec5cefddce6216c80